cmd/promtool/tsdb.go

```diff
@@ -692,3 +723,41 @@ func backfillOpenMetrics(path, outputDir string, humanReadable, quiet bool, maxB
 
 	return checkErr(backfill(5000, inputFile.Bytes(), outputDir, humanReadable, quiet, maxBlockDuration))
 }
+
+func displayHistogram(dataType string, datas []int, total int) {
+	slices.Sort(datas)
+	start, end, step := generateBucket(datas[0], datas[len(datas)-1])
+	sum := 0
+	buckets := make([]int, (end-start)/step+1)
+	maxCount := 0
+	for _, c := range datas {
+		sum += c
+		buckets[(c-start)/step]++
+		if buckets[(c-start)/step] > maxCount {
+			maxCount = buckets[(c-start)/step]
+		}
+	}
+	avg := sum / len(datas)
+	fmt.Printf("%s (min/avg/max): %d/%d/%d\n", dataType, datas[0], avg, datas[len(datas)-1])
+	maxLen := strconv.Itoa(len(fmt.Sprintf("%d", end)))
```

Comment:
Does this have to be `end + step` rather than just `end`? Not sure, but I see the following if only the last upper bound has a greater length:

```
bytes per histogram chunk (min/avg/max): 491/818/862
[401, 500]:    2 
[501, 600]:    1 
[601, 700]:    0 
[701, 800]:  222 
[801, 900]: 2162 #
[901, 1000]:    0 
```

Reply:
yes you are right, besides I also split the `maxLen` to `maxLeftLen` and `maxRightLen`, because if only use the `maxLen`, the output will be like following in your case, which has an extra whitespace on the left.
```
bytes per histogram chunk (min/avg/max): 491/818/862
[ 401,  500]:    2 
[ 501,  600]:    1 
[ 601,  700]:    0 
[ 701,  800]:  222 
[ 801,  900]: 2162 #
[ 901, 1000]:    0 
```

@beorn7